src/daemon/daemon.cpp

```diff
@@ -121,11 +121,6 @@ bool Daemon::activate(const InterfaceConfig& config) {
     }
   }
 
-  // Configure routing for excluded addresses.
-  for (const QString& i : config.m_excludedAddresses) {
-    addExclusionRoute(IPAddress(i));
```

Comment:
Should Daemon.addExclusionRoute, delExclusionRoute and m_excludedAddrSet also be removed?

Reply:
I had the same question but Naomi confirmed that for now those should stay. From @oskirby:

> I think the long term goal would be to remove the exclusion routing code completely, but I think this particular function is still needed on MacOS for the Wireguard server address. The goal here is to addres issue with the smallest amount of change

Reply:
Sure. Perhaps open a ticket to remove additional code later?

Reply:
BTW, I think macOS and Windows use MacosRouteMonitor and WindowsRouteMontitor's addExclusionRoute for excluding the VPN server addresss, which is different from the the Daemon's addExclusionRoute.

Reply:
The following calls from `MacosRouteMonitor` 

```c++
  bool addExclusionRoute(const IPAddress& prefix);
  bool deleteExclusionRoute(const IPAddress& prefix);
  void flushExclusionRoutes();
```
are referenced in `wireguardutilsmacos` so I'm inclined to leave it as is for now. I have a follow up issue here: https://mozilla-hub.atlassian.net/browse/VPN-5649

Reply:
To shed a bit of light on this question. The exclusion routing code in `MacosRouteMonitor` and `WindowsRouteMonitor` are the platform specific logic to interface with the routing table, and the Daemon's methods are wrappers around them which provide reference counting.

The reference counting was important for doing server switching. The gist of how server switching is that we start a connection to a new peer and then drop the connection to the old peer afterwards. If the client specifies an exclusion route then the sequence of events the route monitor sees would look like:
```
Exclude captive portal (old peer)
... wireguard handshake with old peer ...
... some time passes ...
... old peer starts to fail ...
Exclude captive portal exclusion (new peer)
... wireguard handshake with new peer ...
Remove captive portal exclusion (old peer)
```

However, if we are removing the ability for the client to send an exclusion route, then I don't think the reference counting is necessary anymore, and the platform's `WireguardUtils::updatePeer()` methods can just invoke the route monitor directly as needed.

So, the long TLDR is that I think we can remove the daemon's exclusion routing methods in this PR.

Reply:
Exclusion routes are needed for the VPN server, so those can't be removed.